homeassistant/components/myuplink/switch.py

```diff
@@ -0,0 +1,115 @@
+            icon="mdi:water-alert-outline",
+        ),
+    },
+}
+
+
+def get_description(device_point: DevicePoint) -> SwitchEntityDescription | None:
+    """Get description for a device point.
+
+    Priorities:
+    1. Category specific prefix e.g "NIBEF"
+    2. Default to None
+    """
+    prefix, _, _ = device_point.category.partition(" ")
+    description = CATEGORY_BASED_DESCRIPTIONS.get(prefix, {}).get(
+        device_point.parameter_id
+    )
+
+    return description
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up myUplink switch."""
+    entities: list[SwitchEntity] = []
+    coordinator: MyUplinkDataCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    # Setup device point sensors
+    for device_id, point_data in coordinator.data.points.items():
+        for point_id, device_point in point_data.items():
+            if find_matching_platform(device_point) == Platform.SWITCH:
+                description = get_description(device_point)
+
+                entities.append(
+                    MyUplinkDevicePointSwitch(
+                        coordinator=coordinator,
+                        device_id=device_id,
+                        device_point=device_point,
+                        entity_description=description,
+                        unique_id_suffix=point_id,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class MyUplinkDevicePointSwitch(MyUplinkEntity, SwitchEntity):
+    """Representation of a myUplink device point sensor."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: SwitchEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        # Internal properties
+        self.point_id = device_point.parameter_id
+        self._attr_name = device_point.parameter_name
+
+        if entity_description is not None:
+            self.entity_description = entity_description
+
+    @property
+    def is_on(self) -> bool:
+        """Switch state value."""
+        device_point = self.coordinator.data.points[self.device_id][self.point_id]
+        return int(device_point.value) != 0
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on switch."""
+        await self._async_turn_switch(1)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off switch."""
+        await self._async_turn_switch(0)
+
+    async def _async_turn_switch(self, mode: int) -> None:
+        """Set switch mode."""
+        await self.coordinator.api.async_set_device_points(
```

Comment:
Yes, We do a response.raise_for_status() after every request in the lib. Normally there should not be any exceptions here. I can think of one, an error 409 - Conflict when the turn_on service is sent to a switch that is aleady on. This could happen even if we check the state here. Someone else could have toggled the switch in the myuplink app or on the web-site. The state in HA is updated just by polling every minute.

Reply:
No. We need to catch the expected exception and raise `HomeAssistantError`. Preferably the library should raise library specific exceptions that we can use here. For now we can catch `aiohttp.ClientError`.